peachjam/models/core_document.py

```diff
@@ -726,6 +727,19 @@ def extract_citations(self):
         self.delete_citations()
         return citation_analyser.extract_citations(self)
 
+    def extract_citation_contexts(self):
+        """Extract citation contexts for this document. This will extract contexts for all citations
+        in the document, and update the ExtractedCitationContext objects.
+        """
+        from peachjam.analysis.citations import citation_analyser
+
+        self.delete_citation_contexts()
+        citation_analyser.update_citation_contexts(self)
```

Comment:
first check if self is the latest expression

Reply:
i've added a model method/property  `is_latest_expression` that returns true if the document is the latest expression. we do the check in the bg task